runtime/src/iree/hal/command_buffer.h

```diff
@@ -233,6 +233,15 @@ enum iree_hal_collective_kind_e {
   //   ncclAllReduce
   IREE_HAL_COLLECTIVE_KIND_ALL_REDUCE,
 
+  // Gathers |element_count| elements of the specified type in |recv_binding| by
+  // sourcing N parts of |element_count|/N elements, one from the |send_binding|
+  // of each rank, and concatenating them.
+  //
+  // |param|: unused
+  // |send_binding|: local elements to split and send to all ranks
+  // |recv_binding|: concatenated results from all ranks
+  IREE_HAL_COLLECTIVE_KIND_ALL_TO_ALL,
```

Comment:
fyi: changing HAL enum ordering breaks binary compatibility - since no programs are really working with collectives today this particular case is fine but in the future we'd need to preserve enum ordering

Reply:
Understood, thanks. I was wondering about that. I didn't want to be the first one to break the alphabetical ordering.